packages/wallet/src/util.ts

```diff
@@ -1 +1,136 @@
+      } catch (error) {
+        logger.error(error);
+        throw error;
+      }
+    },
+    getChangeAddress: async (): Promise<Cbor> => {
+      logger.debug('getting changeAddress');
+
+      const changeAddresss = wallet.addresses$.value?.find((e) => e.type === 1);
+
+      if (!changeAddresss) {
+        throw new Error('could not find a change address');
+      }
+      return changeAddresss.address.toString();
+      // return cbor.encode(changeAddresss?.address.toString).toString('hex');
+    },
+    getNetworkId: async (): Promise<Cardano.NetworkId> => {
+      logger.debug('getting networkId');
+      return Promise.resolve(props.networkId);
+    },
+    /* getRewardAddresses: async (): Promise<Cbor[]> => {
+      logger.info('getting reward addresses');
+
+      const rewardAccounts = await firstValueFrom(wallet.delegation.rewardAccounts$);
+      return rewardAccounts.map((a) => cbor.encode(a.address.toString).toString('hex'));
+    },*/
+    getUnusedAddresses: async (): Promise<Cbor[]> => {
+      logger.info('getting unused addresses');
+      try {
+        const addresses = await firstValueFrom(wallet.addresses$);
+        const cardanoAddresses = addresses.map((addr) => addr.address as unknown as string);
+        if (!cardanoAddresses) {
+          throw new ApiError(400, 'Could not find any addresses');
+        }
+        return Promise.resolve(cardanoAddresses);
+      } catch (error) {
+        logger.error(error);
+        throw error;
+      }
+    },
+    /* getUsedAddresses: async (_paginate?: Paginate): Promise<Cbor[]> => {
+      const returnedAddresses = [];
+      if (_paginate && _paginate.limit > returnedAddresses.length)
+        throw new PaginateError(returnedAddresses.length, 'Pagination limit exceeds returned addresses');
+      // Otherwise slice the addresses by page/limit
+      return Promise.resolve(['usedAddresses']);
+    },*/
+    getUtxos: async (amount?: Cbor, _paginate?: Paginate): Promise<Cardano.Utxo[] | undefined> => {
+      await firstValueFrom(wallet.utxo.available$);
+      const allUtxos = wallet.utxo.available$.value;
+      if (!allUtxos) return Promise.resolve([]);
+
+      if (amount) {
+        const filteredByAmount = allUtxos
+          .sort((a, b) => {
+            if (a[1].value > b[1].value) return 1;
+            if (b[1].value > a[1].value) return -1;
+            return 0;
+          })
+          .reduce((utxos: Cardano.Utxo[], currentUtxo: Cardano.Utxo) => {
+            const currentSum = utxos.reduce(
+              (sumCalculation, currentUtxoValue) => currentUtxoValue[1].value.coins + sumCalculation,
+              0n
+            );
+            if (currentSum < BigInt(amount)) return [...utxos, currentUtxo];
+            return utxos;
+          }, [] as Cardano.Utxo[]);
+        logger.info('filteredByAmount', filteredByAmount);
+      }
+
+      // TODO: determine how to paginate effectively
```

Comment:
You can use rxjs `skip(n)` and `take(n)` operators before converting it to promise, e.g.
```
await firstValueFrom(wallet.utxo.available$.pipe(skip(0), take(5));
```
You should also apply the same technique converting from core types to cbor through `coreToCsl` util

Reply:
Actually that won't work, as emitted value is an array. You should instead do something like `wallet.utxo.available$.pipe(map(utxo => utxo.slice(...)));`